### PR TITLE
Prevent delete/disable CMS pages used as default pages

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

This does NOT fixes the problem in #382 directly. Disabled pages are still listed in dropdown, but it prevents to disable/delete them.

### Fixed Issues (if relevant)

1. Closes OpenMage/magento-lts#382

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/locale/en_US/Mage_Cms.csv [hunk 1]
```diff
@@ -27,8 +27,10 @@
 "CMS Static Block","CMS Static Block"
 "CMS Static Block Default Template","CMS Static Block Default Template"
 "Cannot create new directory.","Cannot create new directory."
+"Cannot delete page, it is used in "%s".","Cannot delete page, it is used in "%s"."
 "Cannot delete directory %s.","Cannot delete directory %s."
 "Cannot delete root directory %s.","Cannot delete root directory %s."
+"Cannot disable page, it is used in "%s".","Cannot disable page, it is used in "%s"."
```
Comment (luigifab) [inline app/locale/en_US/Mage_Cms.csv:33]: `"Cannot disable page, it is used in ""%s"".","Cannot disable page, it is used in ""%s""."`

Comment (sreichel) [reply app/locale/en_US/Mage_Cms.csv:33]: Is it necessary? Tested it and lgtm?

Comment (luigifab) [reply app/locale/en_US/Mage_Cms.csv:33]: Yes it works, but I think in docs of CSV files you need to escape them.

Comment (fballiano) [reply app/locale/en_US/Mage_Cms.csv:33]: AFAIK it is necessary to doubloe quote quotes in CSV

Comment (sreichel) [reply app/locale/en_US/Mage_Cms.csv:33]: Can you make pls a "suggestion"?

File: app/locale/en_US/Mage_Cms.csv [hunk 2]
```diff
@@ -27,8 +27,10 @@
 "CMS Static Block","CMS Static Block"
 "CMS Static Block Default Template","CMS Static Block Default Template"
 "Cannot create new directory.","Cannot create new directory."
+"Cannot delete page, it is used in "%s".","Cannot delete page, it is used in "%s"."
```
Comment (fballiano) [inline app/locale/en_US/Mage_Cms.csv:30]:
```suggestion
"Cannot delete page, it is used in ""%s"".","Cannot delete page, it is used in ""%s""."
```

File: app/locale/en_US/Mage_Cms.csv (hunk 1 above)
Comment (fballiano) [inline app/locale/en_US/Mage_Cms.csv:33]:
```suggestion
"Cannot disable page, it is used in ""%s"".","Cannot disable page, it is used in ""%s""."
```

File: app/code/core/Mage/Cms/Model/Resource/Page.php [hunk 3]
```diff
@@ -71,6 +83,13 @@ protected function _beforeSave(Mage_Core_Model_Abstract $object)
             $object->setData($field, $this->formatDate($value));
         }
 
+        if (!$object->getIsActive() && ($isUsedInConfig = $this->getIsUsedInStoreConfig($object))) {
+            $object->setIsActive(true);
+            Mage::getSingleton('adminhtml/session')->addWarning(
+                Mage::helper('cms')->__('Cannot disable page, it is used in %s.', implode(', ', $isUsedInConfig))
```
Comment (sreichel) [inline app/code/core/Mage/Cms/Model/Resource/Page.php:89]:
```suggestion
                Mage::helper('cms')->__('Cannot disable page, it is used in "%s".', implode(', ', $isUsedInConfig))
```

**Pull request closed without merging**